azure-kusto-ingest/azure/kusto/ingest/base_ingest_client.py

```diff
@@ -0,0 +1,107 @@
+import os
+import tempfile
+import time
+import uuid
+from abc import ABCMeta, abstractmethod
+from copy import copy
+from enum import Enum
+from gzip import GzipFile
+from io import TextIOWrapper, BytesIO
+from typing import TYPE_CHECKING, Union, IO, AnyStr
+
+from .descriptors import FileDescriptor, StreamDescriptor
+from .ingestion_properties import DataFormat, IngestionProperties
+
+if TYPE_CHECKING:
+    import pandas
+
+
+class IngestionStatus(Enum):
+    """
+    The ingestion was queued.
+    """
+
+    PENDING = "PENDING"
+    """
+    The ingestion was successfully streamed
+    """
+    SUCCESS = "SUCCESS"
+
+
+# TODO - do we want to return an array of statuses, to match java/c#?
+# TODO - do we want to add fields like database, table, etc like java/c#?
+class IngestionResult:
```

Comment:
Done

What about other parameters that we have in java?
table, database, ingestionSourcePath

Reply:
It's all about correlating the requests from client to the service, ingestionSourcePath is useful when we create the blob so it easier to look for it directly, but we can achieve this if the sourceId is passed correctly. 
The baseline should be .Net SDK but you don't have to do it now.

Reply:
I looked at c#, and we don't have all of the baseline now but I added table and database.

We also had a comment in the code that we want to expose IngestionBlobInfo to the user, so I took this opportunity to add it to the result. Tell me what do you think

Reply:
IngestionBlobInfo already contains the sourceId, database and table, so I don't see a reason to have these props twice. 
I'd added the blobUrl instead of blobInfo, but your call..

Reply:
I replaced it with blobUrl to match the c# better.
I also added docs. WDYT?

Reply:
Cool thanks